### DISCUSSION TITLE
Detect when running component tests vs integration tests

### DISCUSSION THREAD

**Discussion by rpeach** [cypress-io/cypress#17659] [Component Testing] [answered]
Hi all,

I have some global `beforeEach`s in my `cypress/support/index.js` file. I want to avoid running some of these in my component tests. Is there a good way to detect whether i'm running component tests (`cypress open-ct`) vs integration tests (`cypress open`)? 

I couldn't find any built-in environment variables or utility functions, all i found was `isBrowser()`

**Comment by rpeach** [comment, marked as answer]
Found the answer to my own question: [`Cypress.testingType`](https://docs.cypress.io/api/cypress-api/testing-type)

**Reply by JessicaSachs** [reply]
We will be **requiring** separate `supportFiles` in 9.x or 10.x to solve this kind of an issue.

The recommended approach is to use the following config and create a shared module between the two support files if you need to share code.

##### `cypress.json`
```json
{
  "e2e": {
    "supportFile": "cypress/support/e2e.js"
  },
  "component": {
    "supportFile": "cypress/support/component.js",
    "testFiles": "**/*.spec.*",
    "componentFolder": "src"
  }
}
```

##### `support/e2e.js`

```js
import shared from './shared-support'

beforeEach(() => {
  // e2e stuff
  shared.someTask()
})
```

##### `support/component.js`

```js
import shared from './shared-support'

beforeEach(() => {
  shared.someTask()
})
```


#### `support/shared.js`

```js
export function someTask() {/* ... */}
```

👍 Hope this helps!
